### ISSUE TITLE
Continuous Deployment silently failing with 0.23.1

### ISSUE THREAD

**Issue by npaufler** [Shopify/shipit-engine#805]
One of our stacks is configured for continuous deployment which worked fine under 0.20 and 0.21, but after upgrading to 0.23 (and then 0.23.1), continuous deployment seems to have silently stopped working.

I can find no errors in the production.log or sidekiq.log, and jobs seem to be enqueuing normally. Manual deployments all continue to work fine.

If I look at the shipit.stacks table in our database, things look normal - pulling out some things that look related:
``
lock_reason: NULL
tasks_count: 1241
continuous_deployment: 1
undeployed_commits_count: 1
lock_author_id: NULL
ignore_ci: 0
continuous_delivery_delayed_since: NULL
locked_since: NULL
merge_queue_enabled: 0
last_deployed_at: 2018-06-20 01:47:52
``

With nothing obviously erroring I'm at a bit of a loss as to figure out why it's stopped working. Any suggestions on what I could check?

Thanks!

**Comment by byroot** [comment]
I'd suggest to walk down `ContinuousDeliveryJob#perform`: https://github.com/Shopify/shipit-engine/blob/42b6d23383f89a5023452ee6212eb65d1f41700a/app/jobs/shipit/continuous_delivery_job.rb#L7-L11

and then `Stack#trigger_continuous_delivery`: https://github.com/Shopify/shipit-engine/blob/42b6d23383f89a5023452ee6212eb65d1f41700a/app/models/shipit/stack.rb#L147-L164

There is a handful of bail out conditions there, if you check them using a Rails console you should be able to identify which one cause problem.

**Comment by charles-ferguson** [comment]
Often when running those commands manually a deploy did work. Our theory was that either they were temporarily failing due to the deploy interval or CI checks not being complete and there was no mechanism to reschedule the job at that point. Is that correct?

In order to work around this issue, we started a `cron` job to run the `cron::minutely` task. Is this an expected workaround?

**Comment by byroot** [comment]
> Is this an expected workaround?

Oh yes. `rake cron:minutely` has to be ran every minutes, continuous delivery and the merge queue absolutely rely on it. So it's not even a workaround, it's the expected behavior.

**Comment by stantona** [comment]
Is this documented anywhere? If not, shout it be?

**Comment by byroot** [comment]
It's indeed missing from https://github.com/Shopify/shipit-engine/blob/master/docs/setup.md

Setups docs are quite a shitshow TBH...

**Comment by sgringwe** [comment]
I believe we also just ran into this issue after upgrading from 0.20.1 to 0.25. Previously, continuous deployment would work, and now it does not always work. I believe it did work once so far, 2 hours ago, but has not done anything since even as new commits come in.

We have actually never had the mentioned cron job running in our deployment of shipit. Maybe there is a change since 0.20.1 that requires the cron task to run minutely.

Either way, is the recommendation to run that rake task every minute to resolve?

**Comment by casperisfine** [comment]
Yes. Shipit always expected that task to be added in cron. I'll blame the setup docs...
